### PR TITLE
OCPQE-18689: install IPI cluster with different disk type on azure

### PR DESCRIPTION
**Pull request by jinyunma**
automate test cases:
[OCP-68596 - [ipi-on-azure] install cluster with disk type Premium_LRS](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-68596)
[OCP-68597 - [ipi-on-azure] install cluster with disk type Standard_LRS on compute node](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-68597)
[OCP-68598 - [ipi-on-azure] install cluster with disk type StandardSSD_LRS](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-68598)

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-premium-lrs-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-premium-lrs-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-worker-standard-lrs-f14 periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-standardssd-lrs-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-azure-mag-ipi-fips-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-standardssd-lrs-f14 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-azure-mag-ipi-fips-f1

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-azure-mag-ipi-fips-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-azure-mag-ipi-fips-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-standardssd-lrs-f14 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-azure-mag-ipi-fips-f14

**Comment by jinyunma** [comment]
1. periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-worker-standard-lrs-f14, [post-check](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47382/rehearse-47382-periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-premium-lrs-f14/1744223345898950656/artifacts/azure-ipi-disktype-premium-lrs-f14/cucushift-installer-check-azure-disktype/build-log.txt)
```
platform:
  azure:
    region: southcentralus
    baseDomainResourceGroupName: os4-common
    defaultMachinePlatform:
      osDisk:
        diskType: Premium_LRS
```
2. periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-worker-standard-lrs-f14, [post-check ](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47382/rehearse-47382-periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-worker-standard-lrs-f14/1744248042346778624/artifacts/azure-ipi-disktype-worker-standard-lrs-f14/cucushift-installer-check-azure-disktype/build-log.txt)
```
controlPlane:
  architecture: amd64
  name: master
  platform:
    azure:
      type: null
compute:
- architecture: amd64
  name: worker
  replicas: 3
  platform:
    azure:
      type: Standard_D4s_v3
      osDisk:
        diskType: Standard_LRS
``` 
3. periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-standardssd-lrs-f14, [post-check](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47382/rehearse-47382-periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-standardssd-lrs-f14/1744305135737114624/artifacts/azure-ipi-disktype-standardssd-lrs-f14/cucushift-installer-check-azure-disktype/build-log.txt)
```
controlPlane:
  architecture: amd64
  name: master
  platform:
    azure:
      type: null
      osDisk:
        diskType: StandardSSD_LRS
compute:
- architecture: amd64
  name: worker
  replicas: 3
  platform:
    azure:
      type: Standard_D4s_v3
      osDisk:
        diskType: StandardSSD_LRS
````
4. rehearse-47382-periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-azure-mag-ipi-fips-f14, [post-check ](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47382/rehearse-47382-periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-azure-mag-ipi-fips-f14/1744305135795834880/artifacts/azure-mag-ipi-fips-f14/cucushift-installer-check-azure-disktype/build-log.txt)

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei to review

File: ci-operator/step-registry/cucushift/installer/check/azure/disktype/cucushift-installer-check-azure-disktype-commands.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+fi
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID "${SHARED_DIR}/metadata.json")
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+config_master_disk_type=$(yq-go r "${INSTALL_CONFIG}" 'controlPlane.platform.azure.osDisk.diskType')
```
[the remark this answers is not in the document]
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/disktype/cucushift-installer-check-azure-disktype-commands.sh]: good points, updated.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-worker-standard-lrs-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-worker-standard-lrs-f14

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml [hunk 2]
```diff
@@ -70,6 +70,37 @@ tests:
...
+    - chain: cucushift-installer-check-cluster-health
+    workflow: cucushift-installer-rehearse-azure-ipi-disktype
+- as: azure-ipi-disktype-standardssd-lrs-f28
+  cron: 54 6 12 * *
+  steps:
+    cluster_profile: azure-qe
+    env:
+      AZURE_COMPUTE_DISK_TYPE: StandardSSD_LRS
+      AZURE_CONTROL_PLANE_DISK_TYPE: StandardSSD_LRS
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
```
Comment (gpei) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml:90]: @jinyunma could we also add `AZURE_DEFAULT_MACHINE_DISK_TYPE` specified here to test another check point, which is when 

> AZURE_DEFAULT_MACHINE_DISK_TYPE
> AZURE_COMPUTE_DISK_TYPE
> AZURE_CONTROL_PLANE_DISK_TYPE

all configured in install-config, the type set in AZURE_COMPUTE_DISK_TYPE and AZURE_CONTROL_PLANE_DISK_TYPE should be used.

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml [hunk 3]
```diff
@@ -70,6 +70,37 @@ tests:
...
+    test:
+    - chain: cucushift-installer-check-cluster-health
+    workflow: cucushift-installer-rehearse-azure-ipi-disktype
+- as: azure-ipi-disktype-worker-standard-lrs-f28
+  cron: 55 9 4 * *
+  steps:
+    cluster_profile: azure-qe
+    env:
+      AZURE_COMPUTE_DISK_TYPE: Standard_LRS
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
```
Comment (gpei) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml:100]: so `Standard_LRS` only could be used for compute node, but not masters?

Comment (jinyunma) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml:100]: yes, it only supports StandardSSD_LRS and Premium_LRS on master

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml (hunk 2 above)
Comment (jinyunma) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.14.yaml:91]: good points, updated

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-disktype-standardssd-lrs-f14

**Pull request merged**